File: problems/Python/14. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""

        for i, c in enumerate(strs[0]):
            for s2 in strs[1:]:
                if i >= len(s2) or s2[i] != c:
                    return strs[0][:i]
        
        return strs[0]

File: problems/Rust/14. Longest Common Prefix.rs
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        if strs.is_empty() {
            return String::new();
        }
        
        let first = strs[0].as_bytes();
        
        for (i, &c) in first.iter().enumerate() {
            for s2 in &strs[1..] {
                if i >= s2.len() || s2.as_bytes()[i] != c {
                    return strs[0][..i].to_string();
                }
            }
        }
        
        strs[0].clone()
    }
}